analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -42,6 +44,10 @@ public class SonarAnalysisContextBase
 
 public abstract class SonarAnalysisContextBase<TContext> : SonarAnalysisContextBase
 {
+    private readonly IGlobPatternMatcher globPatternMatcher;
+
+    private readonly ConditionalWeakTable<Compilation, ConcurrentDictionary<string, bool>> fileCache = new();
```

Comment:
Put it inside the SonarAnalysisContextBase non generic class together with the other caches.
WHY? First of all, it needs to be static so all instances of the Contexts use the same cache.
WHY move and not just make it static here? Because Generic types don't share static fields. Take a look at S2743 for more info.